Apps/W1/Shopify/app/src/Companies/Codeunits/ShpfyCompByTaxId.Codeunit.al

```diff
@@ -0,0 +1,47 @@
+namespace Microsoft.Integration.Shopify;
+//JZA: Task 3 Tax ID
+/// <summary>
+/// Codeunit ShoShpfypify Comp. By Email/Phone (ID 30304) implements Interface Shpfy ICompany Mapping.
+/// </summary>
+codeunit 30366 "Shpfy Comp. By Tax Id" implements "Shpfy ICompany Mapping"
+{
+    Access = Internal;
+
+    internal procedure DoMapping(CompanyId: BigInteger; ShopCode: Code[20]; TemplateCode: Code[20]; AllowCreate: Boolean): Code[20]
+    var
+        ShopifyCompany: Record "Shpfy Company";
+    begin
+        ShopifyCompany.SetAutoCalcFields("Customer No.");
+        if ShopifyCompany.Get(CompanyId) then begin
+            if not IsNullGuid(ShopifyCompany."Customer SystemId") then begin
+                ShopifyCompany.CalcFields("Customer No.");
+                if ShopifyCompany."Customer No." = '' then begin
+                    Clear(ShopifyCompany."Customer SystemId");
+                    ShopifyCompany.Modify();
+                end else
+                    exit(ShopifyCompany."Customer No.");
+            end;
+            exit(CreateCompany(CompanyId, ShopCode, TemplateCode, AllowCreate));
+        end else
+            exit(CreateCompany(CompanyId, ShopCode, TemplateCode, AllowCreate));
+        exit('');
```

Comment:
this exit will never be reached? It looks like the CreateCompany should be the last exit and is not needed in both paths there?

Reply:
removed